src/tutorials/0005-regular-files-api/06.js

```diff
@@ -66,19 +76,21 @@ const validate = async (result, ipfs) => {
   if (JSON.stringify(expectedResults) === JSON.stringify(result)) {
     return {
       success: 'Success!',
-      logDesc: "Here are the results returned by the `ls` method for the top-level directory. Notice that there are new fields here that we didn't see in the data returned by the `add` method. Also, take a look at how the `hash` and `path` values now differ. The `hash` for each file is the CID of the file itself, while the the `path` is the CID of the top-level directory followed by the filename.",
-      log: result
+      logDesc: "Here are the results returned by the `ls` method for the top-level directory. Notice that there are new fields here that we didn't see in the data returned by the `add` method. Also, take a look at how the `cid` and `path` values now differ. The `cid` for each file is the CID of the file itself, while the the `path` is the CID of the top-level directory followed by the filename.",
+      log: result.map(utils.format.ipfsObject)
     }
   } else {
     return { fail: `Something seems to be wrong. Please click "Reset Code" and try again, taking another look at the instructions and editing only the portion of code indicated. Feeling really stuck? You can click "View Solution" to see our suggested code.` }
   }
 }
 
 const code = `/* global ipfs */
+const all = require('it-all')
+
 const run = async (files) => {
   const fileObjectsArray = files.map((file) => { return { path: file.name, content: file }})
-  const addedFiles = await ipfs.add(fileObjectsArray, { wrapWithDirectory: true })
-  const directoryCID = addedFiles[addedFiles.length - 1].hash
+  const addedFiles = await all(ipfs.add(fileObjectsArray, { wrapWithDirectory: true }))
+  const directoryCID = addedFiles[addedFiles.length - 1].cid
```

Comment:
If we keep the `it-` modules then it might be more appropriate to use `it-last` here.

Reply:
I think it is ok to keep it as is since we only have one extra LOC and we don't really envision any performance issues in the tutorials.

What do you think @terichadbourne?